chain/transaction.go

```diff
@@ -359,6 +362,92 @@ func (t *Transaction) Marshal(p *codec.Packer) error {
 	return t.marshal(p)
 }
 
+func (t *Transaction) MarshalJSON() ([]byte, error) {
+	type txJSON struct {
+		ID      ids.ID `json:"id"`
+		Actions []byte `json:"actions"`
+		Auth    []byte `json:"auth"`
```

Comment:
These types should be `codec.Bytes` so that marshalling them to hex is handled by `json.Marshal`